epochStart/metachain/rewardsCreatorProxy.go

```diff
@@ -0,0 +1,188 @@
+	return rcp.rc.GetProtocolSustainabilityRewards()
+}
+
+// GetLocalTxCache proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) GetLocalTxCache() epochStart.TransactionCacher {
+	return rcp.rc.GetLocalTxCache()
+}
+
+// CreateMarshalizedData proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) CreateMarshalizedData(body *block.Body) map[string][][]byte {
+	return rcp.rc.CreateMarshalizedData(body)
+}
+
+// GetRewardsTxs proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) GetRewardsTxs(body *block.Body) map[string]data.TransactionHandler {
+	return rcp.rc.GetRewardsTxs(body)
+}
+
+// SaveTxBlockToStorage proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) SaveTxBlockToStorage(metaBlock *block.MetaBlock, body *block.Body) {
+	rcp.rc.SaveTxBlockToStorage(metaBlock, body)
+}
+
+// // RemoveBlockDataFromPools proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) DeleteTxsFromStorage(metaBlock *block.MetaBlock, body *block.Body) {
+	rcp.rc.DeleteTxsFromStorage(metaBlock, body)
+}
+
+// RemoveBlockDataFromPools proxies the same method of the configured rewardsCreator instance
+func (rcp *rewardsCreatorProxy) RemoveBlockDataFromPools(metaBlock *block.MetaBlock, body *block.Body) {
+	rcp.rc.RemoveBlockDataFromPools(metaBlock, body)
+}
+
+// IsInterfaceNil returns nil if the underlying object is nil
+func (rcp *rewardsCreatorProxy) IsInterfaceNil() bool {
+	return rcp == nil
+}
+
+func (rcp *rewardsCreatorProxy) changeRewardCreatorIfNeeded(epoch uint32) error {
+	rcp.mutRc.Lock()
+	defer rcp.mutRc.Unlock()
+
+	switch rcp.configuredRC {
+	case rCreatorV1:
+		if epoch > rcp.epochEnableV2 {
+			return rcp.switchToRewardsCreatorV2()
+		}
+	case rCreatorV2:
+		if epoch <= rcp.epochEnableV2 {
+			return rcp.switchToRewardsCreatorV1()
+		}
+	default:
+		return epochStart.ErrInvalidRewardsCreatorProxyConfig
+	}
+
+	return nil
+}
+
+// to be called under locked mutex
+func (rcp *rewardsCreatorProxy) switchToRewardsCreatorV1() error {
+	argsV1 := ArgsNewRewardsCreator{
+		BaseRewardsCreatorArgs: rcp.args.BaseRewardsCreatorArgs,
+	}
+
+	rcV1, err := NewEpochStartRewardsCreator(argsV1)
+	if err != nil {
+		return err
+	}
+
+	rcp.rc = rcV1
+	rcp.configuredRC = rCreatorV1
+
+	log.Info("rewardsCreatorProxy.switchToRewardsCreatorV1")
+
+	return nil
+}
+
+// to be called under locked mutex
+func (rcp *rewardsCreatorProxy) switchToRewardsCreatorV2() error {
+	argsV2 := RewardsCreatorArgsV2{
+		BaseRewardsCreatorArgs: rcp.args.BaseRewardsCreatorArgs,
+		StakingDataProvider:    rcp.args.StakingDataProvider,
+		EconomicsDataProvider:  rcp.args.EconomicsDataProvider,
+		TopUpRewardFactor:      rcp.args.TopUpRewardFactor,
+		TopUpGradientPoint:     rcp.args.TopUpGradientPoint,
+	}
+
+	rcV2, err := NewEpochStartRewardsCreatorV2(argsV2)
```

Comment:
that's what I did first, but didn't want to drag the original instance after the v2 enable epoch, since anyway it is easy to create.
If we change again, then we would need to drag 2 other instances.